network/codec/cbor/encode.go

```diff
@@ -182,26 +178,18 @@ func switchv2what(v interface{}) (string, error) {
 	return what, nil
 }
 
-func v2envEncode(v interface{}, via string) ([]byte, uint8, error) {
+func v2envelopeCode(v interface{}) (string, uint8, error) {
```

Comment:
mind add some comments for this function. Am not 100% sure what is it doing.

Reply:
@vishalchangrani thanks for the comment. As requested I added extra comments. Please let me know if it's clearer now :-)